examples/platform/silabs/shell/tracing/TracingShellCommands.cpp

```diff
@@ -0,0 +1,203 @@
+    }
+    else if (strcmp(str, "TransportSetup") == 0)
+    {
+        return TimeTraceOperation::kTransportSetup;
+    }
+    else if (strcmp(str, "FindOperational") == 0)
+    {
+        return TimeTraceOperation::kFindOperational;
+    }
+    else if (strcmp(str, "CaseSession") == 0)
+    {
+        return TimeTraceOperation::kCaseSession;
+    }
+    else if (strcmp(str, "Sigma1") == 0)
+    {
+        return TimeTraceOperation::kSigma1;
+    }
+    else if (strcmp(str, "Sigma2") == 0)
+    {
+        return TimeTraceOperation::kSigma2;
+    }
+    else if (strcmp(str, "Sigma3") == 0)
+    {
+        return TimeTraceOperation::kSigma3;
+    }
+    else if (strcmp(str, "OTA") == 0)
+    {
+        return TimeTraceOperation::kOTA;
+    }
+    else if (strcmp(str, "ImageUpload") == 0)
+    {
+        return TimeTraceOperation::kImageUpload;
+    }
+    else if (strcmp(str, "ImageVerification") == 0)
+    {
+        return TimeTraceOperation::kImageVerification;
+    }
+    else if (strcmp(str, "AppApplyTime") == 0)
+    {
+        return TimeTraceOperation::kAppApplyTime;
+    }
+    else if (strcmp(str, "Bootup") == 0)
+    {
+        return TimeTraceOperation::kBootup;
+    }
+    else if (strcmp(str, "SilabsInit") == 0)
+    {
+        return TimeTraceOperation::kSilabsInit;
+    }
+    else if (strcmp(str, "MatterInit") == 0)
+    {
+        return TimeTraceOperation::kMatterInit;
+    }
+    else if (strcmp(str, "BufferFull") == 0)
+    {
+        return TimeTraceOperation::kBufferFull;
+    }
+    else
+    {
+        return TimeTraceOperation::kNumTraces;
+    }
+}
+
+Engine sShellTracingSubCommands;
+
+/********************************************************
+ * Silabs Tracing shell functions
+ *********************************************************/
+
+CHIP_ERROR TracingHelpHandler(int argc, char ** argv)
+{
+    sShellTracingSubCommands.ForEachCommand(Shell::PrintCommandHelp, nullptr);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR TracingCommandHandler(int argc, char ** argv)
+{
+    if (argc == 0)
+    {
+        return TracingHelpHandler(argc, argv);
+    }
+
+    return sShellTracingSubCommands.ExecCommand(argc, argv);
+}
+
+CHIP_ERROR WatermarksCommandHandler(int argc, char ** argv)
+{
+    CHIP_ERROR error = CHIP_NO_ERROR;
+    if (strcmp(argv[0], "all") == 0)
+    {
+        error = Tracing::Silabs::SilabsTracer::Instance().OutputAllWaterMarks();
+    }
+    else
+    {
+        TimeTraceOperation operation = StringToTimeTraceOperation(argv[1]);
+        error                        = Tracing::Silabs::SilabsTracer::Instance().OutputWaterMark(operation);
+    }
+    return error;
+}
+
+CHIP_ERROR FlushCommandHandler(int argc, char ** argv)
+{
+    CHIP_ERROR error = CHIP_NO_ERROR;
+    if (strcmp(argv[0], "all") == 0)
+    {
+        error = Tracing::Silabs::SilabsTracer::Instance().TraceBufferFlushAll();
+    }
+    else
+    {
+        TimeTraceOperation operation = StringToTimeTraceOperation(argv[1]);
```

Comment:
What happens if the operation doesn't exist in the string to time function?

In practice, it would be interesting to have a second layer of commands that defines all the operations so that they can be printed with a help menu.

Reply:
For the watermarks, it will output an error, for the flush, it won't output anything since it won't find a trace for the kNumTrace event, I'll implement a LisTimeTraceOperations method to display this.